packages/ui-library/src/elements/stepper/index.js

```diff
@@ -0,0 +1,126 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React, { forwardRef, useRef, useEffect, useState, useCallback } from "react";
+import { CheckIcon } from "@heroicons/react/solid";
+
+/**
+ * Step component.
+ *
+ * @param {string} label The step label.
+ * @param {number} index The step index.
+ * @param {boolean} isComplete Is the step complete.
+ * @param {number} currentStep The current step.
+ * @param {function} addStepRef The function to add the step ref.
+ *
+ * @returns {JSX.Element} The step element.
+ */
+const Step = ( { label, index, isComplete, currentStep, addStepRef } ) => {
+	const handleRef = useCallback( ( el ) =>{
+		addStepRef( el, index );
+	}, [] );
+	return (
+		<div
+			ref={ handleRef }
+			className={ classNames( "yst-flex yst-flex-col yst-items-center",
+				currentStep > index + 1 || isComplete ? "yst-step--complete" : "",
+				currentStep === index + 1 ? "yst-step--active" : "" ) }
+		>
+			<div className="yst-step-circle">
+				{ ( currentStep > index + 1 || isComplete ) && <CheckIcon className="yst-w-3" /> }
+
+				{ ! isComplete && currentStep === index + 1 && <svg className="yst-text-primary-500" width="9" height="9" xmlns="http://www.w3.org/2000/svg">
+					<circle cx="4.5" cy="4.5" r="4.5" fill="currentColor" />
+				</svg> }
+			</div>
+			<div className="yst-font-semibold yst-text-xxs yst-mt-3">{ label }</div>
+		</div>
+	);
+};
+
+Step.propTypes = {
+	label: PropTypes.string.isRequired,
+	index: PropTypes.number.isRequired,
+	isComplete: PropTypes.bool.isRequired,
+	currentStep: PropTypes.number.isRequired,
+	addStepRef: PropTypes.func.isRequired,
+};
+
+/**
+ *
+ * @param {number} currentStep The currrent step.
+ * @param {boolean} isComplete Is the step complete.
+ * @param {array} steps The steps names.
+ * @param {string} [className] Optional extra className.
+ * @returns {JSX.Element} The Stepper element.
+ */
+const Stepper = forwardRef( ( { currentStep, isComplete, steps, className = "" }, ref ) => {
+	const [ margins, setMargins ] = useState( {
+		marginLeft: 0,
+		marginRight: 0,
+	} );
+	const stepRef = useRef( [] );
+
+	useEffect( () => {
```

Comment:
Shouldn't this calculation happen before rendering instead of after?
E.g. there can be a point where the rendered thing is wrong for a moment in time, potentially looking funky.

Reply:
the ref Current is empty before render